ocean_provider/utils/datatoken.py

```diff
@@ -67,14 +82,31 @@ def verify_order_tx(
     assert (
         asset_id == remove_0x_prefix(contract.address).lower()
     ), "asset-id does not match the datatoken id."
-    if str(order_log.args.serviceId) != str(service_id):
+    if str(order_log.args.serviceId) != str(service.index):
         raise AssertionError(
             f"The asset id (DID) or service id in the event does "
             f"not match the requested asset. \n"
-            f"requested: (did={did}, serviceId={service_id}\n"
+            f"requested: (did={did}, serviceId={service.index}\n"
             f"event: (serviceId={order_log.args.serviceId}"
         )
 
+    # Check if order expired. timeout == 0 means order is valid forever
+    service_timeout = service.main["timeout"]
+    timestamp_now = datetime.utcnow().timestamp()
+    timestamp_delta = timestamp_now - order_log.args.timestamp
+    logger.debug(
+        f"verify_order_tx: service timeout = {service_timeout}, timestamp delta = {timestamp_delta}"
+    )
+    if service_timeout != 0:
+        if timestamp_delta > service_timeout:
```

Comment:
I think if+if can be transformed to an "and"

Reply:
Done.